### DISCUSSION TITLE
tea without Xcode-cli-tools

### DISCUSSION THREAD

**Discussion by tnxz** [pkgxdev/pkgx#307] [Help] [answered]
Have installed tea with out installing Xcode-cli-tools don't seem to be running into any issues other than occasional popup to install them when something like git or gcc are invoked without wrapping them with tea. This can be solved symlinking them(git, gcc, ..) to tea as described in tool to PATH in the readme.

Will there be any foreseeable problems that might be encountered?

**Comment by mxcl** [comment]
Xcode CLT provides the macOS SDK. If something on mac needs eg `gcc` then sometimes it also need the SDKs.

I would love for there to be a way to provide the SDKs without the Xcode CLT, hopefully we can all figure out a way some day. Also would love to be able to provide something like the SDKs that works in most cases, but we haven't had much luck there yet either.

So, you can `tea +llvm.org +tea.xyz/gx/cc` to try to use our `cc`, `gcc` etc. without Xcode CLT. If you have problems then install the Xcode CLT.

**Reply by tnxz** [reply]
recently tried to compile a project with the same command as you pointed, but it is complaining about `ld`

**Reply by mxcl** [reply]
What's the complaint? We have good experience here.

**Reply by tnxz** [reply]
`./configure ...` prompts no `ld` and install xcode clt prompt

`ln -s $(which tea) ~/.local/bin/ld` still prompts to install `xcode clt`

`which ld` - `~/.local/bin/ld`

`ld` does trigger `~/.tea/.../bin/ld` from `tea -n ld`

`~/.local/bin` is on the PATH

**Comment by tnxz** [comment]
`cc` seems to be triggering popups even on installing tea 
`+llvim.org +tea.xyz/gx/cc`. Am I missing any other package??

**Reply by jhheider** [reply, marked as answer]
As it currently stands, `cc` requires the CLT on darwin: https://github.com/teaxyz/pantry/blob/bdbb2fcecf7bcc5baef86a91b30bcac96da18240/projects/tea.xyz/gx/cc/package.yml#L8-L12

You should be able to use `llvm.org` *in place of* `tea.xyz/gx/cc` in many applications, however our packaged build mechanics generally depend on our `cc` for the SDK reasons. But in any project which you control, you have the ability to use llvm or gcc or something else freely.

**Reply by mxcl** [reply]
If you `+llvm.org` it `cc` should use it, if not it's a bug. Please open a ticket if so.
